### PR TITLE
tests: Test no false overlap with sliced 3d image as framebuffer

### PR DESCRIPTION
**Pull request by aitor-lunarg**
Create renderpass with multiple subpasses that draw to color attachments. These attachments won't overlap since the framebuffer will have different slices of a 3D image as attachments.

Test was requested in #4207. Without #4207 this test fails due to incorrectly reporting attachment overlap.

### CODE REVIEW

File: tests/positive/image_buffer.cpp [hunk 1]
```diff
@@ -2644,3 +2644,113 @@ TEST_F(VkPositiveLayerTest, ImageCompressionControl) {
...
+    auto fb_info = LvlInitStruct<VkFramebufferCreateInfo>();
+    fb_info.renderPass = renderpass;
+    fb_info.attachmentCount = depth_count;
+    fb_info.pAttachments = views;
+    fb_info.width = image_info.extent.width;
+    fb_info.height = image_info.extent.height;
+    fb_info.layers = 1;
+
+    VkFramebuffer framebuffer = VK_NULL_HANDLE;
+    vk::CreateFramebuffer(m_device->handle(), &fb_info, nullptr, &framebuffer);
```
Comment (jeremyg-lunarg) [inline tests/positive/image_buffer.cpp:2734]: vk_testing::Framebuffer?

**Review by jeremyg-lunarg — COMMENTED** [review]
Test looks correct to me. Using the RAII vulkan handle objects would remove the need for manual cleanup code. (this is another thing that is very inconsistently done in existing test cases)

File: tests/positive/image_buffer.cpp [hunk 2]
```diff
@@ -2644,3 +2644,113 @@ TEST_F(VkPositiveLayerTest, ImageCompressionControl) {
...
+    view_info.format = image_info.format;
+    view_info.components.r = VK_COMPONENT_SWIZZLE_R;
+    view_info.components.g = VK_COMPONENT_SWIZZLE_G;
+    view_info.components.b = VK_COMPONENT_SWIZZLE_B;
+    view_info.components.a = VK_COMPONENT_SWIZZLE_A;
+    view_info.subresourceRange = {VK_IMAGE_ASPECT_COLOR_BIT, 0, 1, 0, 1};
+    VkImageView views[depth_count] = {VK_NULL_HANDLE};
+    for (unsigned i = 0; i < depth_count; ++i) {
+        view_info.subresourceRange.baseArrayLayer = i;
+        vk::CreateImageView(m_device->handle(), &view_info, nullptr, &views[i]);
```
[the remark this answers is not in the document]
Comment (aitor-lunarg) [reply tests/positive/image_buffer.cpp]: Thank you for pointing it out! I completely forgot those existed (I tried looking for ImageViewObj. I think the reason for inconsistencies is that we have multiple things that do similar things or more. Which in turn leads to forgetting the base ones).

[the remark this answers is not in the document]
Comment (jeremyg-lunarg) [reply tests/positive/image_buffer.cpp]: yep cleaning up and keeping the tests clean is going to take quite a bit of effort

**Review by Tony-LunarG — APPROVED** [review]
LGTM

**Pull request merged**